### PR TITLE
Fix memory leak in test

### PR DESCRIPTION
**Pull request by kbinias**
This PR fixes the memory leak in the module that initiates all unit tests.

The fix is not critical because the system will deallocate the memory when exiting the function, but limits the number of errors in the AddressSanitizer report: `==56712==ERROR: LeakSanitizer: detected memory leaks`

### CODE REVIEW

**Review by jczaja — DISMISSED** [review]
LGTM

File: paddle/testing/paddle_gtest_main.cc [hunk 1]
```diff
@@ -61,29 +61,30 @@ int main(int argc, char** argv) {
...
 
+  std::string env_string = "--tryfromenv=";
   if (envs.size() > 0) {
-    std::string env_string = "--tryfromenv=";
     for (auto t : envs) {
       env_string += t + ",";
     }
     env_string = env_string.substr(0, env_string.length() - 1);
-    new_argv.push_back(strdup(env_string.c_str()));
+    new_argv.push_back(const_cast<char*>(env_string.c_str()));
```
Comment (luotao1) [inline paddle/testing/paddle_gtest_main.cc]: It is not recommended to use `const_cast`, could you use another way?

Comment (kbinias) [reply paddle/testing/paddle_gtest_main.cc]: I prepared other solution (without const_cast)

**Review by luotao1 — APPROVED** [review]
LGTM

**Pull request merged**